src/parser/jobs/blm/modules/Triple.js

```diff
@@ -35,48 +40,55 @@ export default class Triple extends Module {
 	}
 
 	_onCast(event) {
-		const actionId = event.ability.guid
-
-		// Start tracking
-		if (actionId === ACTIONS.TRIPLECAST.id) {
-			this._active = true
-			this._triple = {
-				start: event.timestamp,
-				end: null,
-				casts: [],
-			}
+		const action = getAction(event.ability.guid)
 
-			this._ctIndex = this.castTime.set('all', 0)
-		}
-
-		if (!this._active || getAction(actionId).autoAttack) {
+		if (!this._active || getAction(action.id).autoAttack) {
 			return
 		}
-
-		// Save the event to the DWT casts
-		this._triple.casts.push(event)
+		//check if Triple window is active
+		if (this._active) {
+			//stop tracking on next GCD after triple is gone
+			if (action.onGcd && this._tripleFlag) {
+				this._stopRecording()
+			} else {
+				this._triple.casts.push(event)
+			}
+		}
 	}
 
 	_onComplete() {
 		// Clean up any existing casts
 		if (this._active) {
-			this._onRemoveTriple()
+			this._stopRecording()
+		}
+	}
+
+	_onApplyTriple(event) {
+		//start tracking
+		this._active = true
+		this._triple = {
+			start: event.timestamp,
+			end: null,
+			casts: [],
 		}
+		this._ctIndex = this.castTime.set('all', 0)
 	}
 
 	_onRemoveTriple() {
-		if (this._active) {
-			this._active = false
-			this._triple.end = this.parser.currentTimestamp
-			this._history.push(this._triple)
+		this._tripleFlag = true
+	}
 
-			this.castTime.reset(this._ctIndex)
-		}
+	_stopRecording() {
+		this._active = false
+		this._tripleFlag = false
+		this._triple.end = this.parser.currentTimestamp
+		this._history.push(this._triple)
+		this.castTime.reset(this._ctIndex)
 	}
 
 	output() {
 		const panels = this._history.map(triple => {
-			const numGcds = triple.casts.filter(cast => !getAction(cast.ability.guid).onGcd).length - 1 // 1 is Triplecast itself
+			const numGcds = triple.casts.filter(cast => !getAction(cast.ability.guid).onGcd).length
```

Comment:
I just realised I do this too but it's technically fine since the action is pre-validated before the event is pushed to the array.